webapp/src/ts/modules/reports/reports.component.ts

```diff
@@ -325,29 +331,34 @@ export class ReportsComponent implements OnInit, AfterViewInit, OnDestroy {
       return;
     }
 
+    const exportFn = function () {
+      this.exportReports();
+    };
     this.globalActions.setLeftActionBar({
+      exportFn: exportFn.bind(this),
       hasResults: this.hasReports,
-      exportFn: this.exportFn.bind({}, this.ngZone, this.exportService, this.filters),
     });
   }
 
-  private exportFn(ngZone, exportService, filters, e) {
-    const exportFilters = _cloneDeep(filters);
+  exportReports() {
+    if (this.isExporting) {
+      return;
+    }
+
+    const exportFilters = _cloneDeep(this.filters);
     ['forms', 'facilities'].forEach((type) => {
       if (exportFilters[type]) {
         delete exportFilters[type].options;
       }
     });
 
-    const $link = $(e.target).closest('a');
-    $link.addClass('mm-icon-disabled');
-    ngZone.runOutsideAngular(() => {
-      setTimeout(() => {
-        $link.removeClass('mm-icon-disabled');
-      }, 2000);
+    // TODO: Improve debounce or make a way to determine when the export has finished.
+    this.isExporting = true;
```

Comment:
I'm thinking of leaving this ToDo on purpose here for future refactoring because the export service is used in several places and better have that work isolated in a different PR. I just improve it a tiny bit.

Reply:
I agree it would be a bit too much to rework within this PR. Do we have an open issue tracking this todo?

Reply:
I didn't find an issue with that, so I created [this one](https://github.com/medic/cht-core/issues/7991). Maybe if you have previous experience doing this and would like to suggest a solution to that issue, it would be awesome! :)